### PR TITLE
Make trading API errors more consistent

### LINKED ISSUE #7477: Error handling very inconsistent in the v2.tradingDataService
Some places we do this: 

t.formatE(err)

Other places we do this:

 errors.Wrap(err, "making edges")

And other places we do this:

fmt.Errorf("could not retrieve market data history for market id: %w", err)   (this latter should be a not found if message).

We should tidy it all up to do t.formatE(err) and correctly use not found errors -  user experience thing so @gordsport maybe do this asap?

### PR DESCRIPTION
**Pull request by zale144**
Resolves #7477

### CODE REVIEW

File: datanode/api/trading_data_v2.go [hunk 1]
```diff
@@ -103,22 +102,22 @@ func (t *tradingDataServiceV2) ListAccounts(ctx context.Context, req *v2.ListAcc
 	defer metrics.StartAPIRequestAndTimeGRPC("ListAccountsV2")()
 	pagination, err := entities.CursorPaginationFromProto(req.Pagination)
 	if err != nil {
-		return nil, apiError(codes.InvalidArgument, ErrMalformedRequest, err)
+		return nil, formatE(err, ErrInvalidPagination.Error())
```
Comment (wwestgarth) [inline datanode/api/trading_data_v2.go:105]: Before this was returning `codes.InvalidArgument`, but now it'll be `code.InternalError` when the error could be something like `pagination limit must be greater than 0`.

File: datanode/broker/buffer_files_event_source.go [hunk 2]
```diff
@@ -143,6 +144,7 @@ func (e *bufferFileEventSource) sendAllRawEventsInFile(ctx context.Context, out
 			}
 		}
 	}
+	return true
```
[the remark this answers is not in the document]
Comment (zale144) [reply datanode/broker/buffer_files_event_source.go:147]: nice catch. bad merge

File: datanode/sqlstore/accounts_test.go [hunk 3]
```diff
@@ -94,6 +95,7 @@ func TestAccount(t *testing.T) {
 		assert.Len(t, accs, 0)
 	})
 
+	// TODO: remove.
```
Comment (ze97286) [inline datanode/sqlstore/accounts_test.go:98]: ?

File: datanode/sqlstore/accounts.go [hunk 4]
```diff
@@ -195,6 +196,7 @@ func (as *Accounts) Query(ctx context.Context, filter entities.AccountFilter) ([
 	return accs, nil
 }
 
+// TODO: remove.
```
[the remark this answers is not in the document]
Comment (guoguojin) [reply datanode/sqlstore/accounts.go:199]: Yes, all the v1 queries have been removed from the API now so anything supporting them should be safe to remove.

**Review by pscott31 — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

File: datanode/sqlstore/accounts_test.go [hunk 5]
```diff
@@ -15,13 +15,14 @@ package sqlstore_test
 import (
 	"testing"
 
+	"github.com/shopspring/decimal"
```
Comment (EVODelavega) [inline datanode/sqlstore/accounts_test.go:18]: nit-pick: generally we tend to group imports as:

```go
import(
    "stdpkg"
    
     "code.vegaprotocol.io/vega/tpkg"
     
     "github.com/third/party/pkg"
)
```

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**